backend/controllers/specializationsControllers.js

```diff
@@ -0,0 +1,149 @@
+const SpecializationsModel = require("../models/SpecializationsModel");
+const User = require("../models/UserModel");
+var mongoose = require("mongoose");
+
+module.exports.add = async (req, res) => {
+  try {
+    const { name, contributed_by, doctor_ids } = req.body;
+    if (!name) {
+      return res.status(400).json({ error: "Name is empty" });
+    }
+    const isValidContributed_by =
+      mongoose.Types.ObjectId.isValid(contributed_by); //true
+    if (!isValidContributed_by) {
+      return res.status(400).json({ error: "Invalid Contributor ID" });
+    }
+
+    // Validate user IDs
+    const contributor = await User.find({
+      _id: contributed_by,
+      isDoctor: false,
+    });
+    if (!contributor) {
+      return res.status(400).json({ error: "User doesn't exist" });
+    }
+    // Validate if each doctor_id is a valid ObjectId
+    for (const id of doctor_ids) {
```

Comment:
doctor_id will be updated from here, in such a way that with this category the newly generated id will be added, for that newly registered user, specialization API will be called from the **register function** and **update profile** function.